src/components/confirm/confirm.component.js

```diff
@@ -44,6 +49,23 @@ const Confirm = ({
   }
 
   const l = useLocale();
+
+  let titleId;
+  let subtitleId;
+  const ariaProps = {
+    "aria-labelledby": ariaLabelledBy,
+    "aria-describedby": ariaDescribedBy,
+    "aria-label": ariaLabel,
+  };
+  if (title && iconType) {
+    titleId = createGuid();
```

Comment:
This will lead to `titleId` and `subtitleId` being a different value on each render. You should use `useRef` in this case like:
const { current: titleId } = useRef(createGuid())